scripts/prepare_sector_network.py

```diff
@@ -1960,14 +2231,74 @@ def get_temp_efficency(
 
 
 def add_EVs(
-    n,
-    avail_profile,
-    dsm_profile,
-    p_set,
-    electric_share,
-    number_cars,
-    temperature,
-):
+    n: pypsa.Network,
+    avail_profile: pd.DataFrame,
+    dsm_profile: pd.DataFrame,
+    p_set: pd.Series,
+    electric_share: pd.Series,
+    number_cars: pd.Series,
+    temperature: pd.DataFrame,
+    spatial: SimpleNamespace,
+    options: dict,
+) -> None:
+    """
+    Add electric vehicle (EV) infrastructure to the network.
+
+    Creates EV batteries, chargers, and optional vehicle-to-grid (V2G) components
+    with temperature-dependent efficiency and demand-side management capabilities.
+
+    Parameters
+    ----------
+    n : pypsa.Network
+        The PyPSA network container object to be modified
+    avail_profile : pd.DataFrame
+        Availability profile for EV charging with snapshots as index and nodes as columns
+    dsm_profile : pd.DataFrame
+        Demand-side management profile defining minimum state of charge
+        with snapshots as index and nodes as columns
+    p_set : pd.Series
+        Base power demand profile for EVs
+    electric_share : pd.Series
+        Share of electric vehicles per node
+    number_cars : pd.Series
+        Number of cars per node
+    temperature : pd.DataFrame
+        Ambient temperature per node and timestamp
+    spatial : SimpleNamespace
+        Spatial configuration containing at least:
+        - nodes: list or Index of node names
+    options : dict
+        Configuration options containing at least:
+        - transport_electric_efficiency: float
+        - transport_heating_deadband_lower: float
+        - transport_heating_deadband_upper: float
+        - EV_lower_degree_factor: float
+        - EV_upper_degree_factor: float
+        - bev_charge_rate: float
+        - bev_charge_efficiency: float
+        - bev_dsm: bool
+        - bev_energy: float
+        - bev_dsm_availability: float
+        - v2g: bool
```

Comment:
This is going to be hard to maintain and should rather be in a single place -> json-schema.